assign_rights/test_helpers.py

```diff
@@ -28,31 +34,103 @@ def add_groupings(count=5):
+
+
 def add_rights_shells(count=15):
-    for x in range(count):
+    def copyright_shell():
+        date_info = set_dates()
+        RightsShell.objects.create(
+            rights_basis="copyright",
+            copyright_status=random.choice(["copyrighted", "public domain", "unknown"]),
+            jurisdiction="us",
+            determination_date=random_date(10, 0),
+            note=random_string(),
+            start_date=date_info[0],
+            end_date=date_info[1],
+            start_date_period=date_info[2],
+            end_date_period=date_info[3],
+            end_date_open=date_info[4],
+        )
+
+    def other_shell():
+        date_info = set_dates()
         RightsShell.objects.create(
-            rights_basis=random.choice([b[0] for b in RightsShell.RIGHTS_BASIS_CHOICES]),
-            copyright_status="copyrighted",
-            determination_date=random_date(),
+            rights_basis=random.choice(["policy", "donor"]),
+            determination_date=random_date(10, 0),
             note=random_string(),
-            start_date=random.choice([None, random_date()]),
-            end_date=random.choice([None, random_date()]),
-            start_date_period=random.randint(0, 10),
-            end_date_period=random.randint(0, 10),
-            end_date_open=random.choice([True, False]),
-            license_terms=None,
-            statute_citation=None)
+            start_date=date_info[0],
+            end_date=date_info[1],
+            start_date_period=date_info[2],
+            end_date_period=date_info[3],
+            end_date_open=date_info[4],
+        )
+
+    def statute_shell():
+        date_info = set_dates()
+        RightsShell.objects.create(
+            rights_basis="statute",
+            jurisdiction="us",
+            determination_date=random_date(10, 0),
+            note=random_string(),
+            start_date=date_info[0],
+            end_date=date_info[1],
+            start_date_period=date_info[2],
+            end_date_period=date_info[3],
+            end_date_open=date_info[4],
+            statute_citation=random_string(),
+        )
+
+    def license_shell():
+        date_info = set_dates()
+        RightsShell.objects.create(
+            rights_basis="license",
+            jurisdiction="us",
+            determination_date=random_date(10, 0),
+            note=random_string(),
+            start_date=date_info[0],
+            end_date=date_info[1],
+            start_date_period=date_info[2],
+            end_date_period=date_info[3],
+            end_date_open=date_info[4],
+            license_terms=random_string(),
+        )
+
+    shell_types = [copyright_shell, other_shell, statute_shell, license_shell]
+    for x in range(count):
+        random.choice(shell_types)()
 
 
 def add_rights_acts(count=5):
     for x in range(count):
+        date_info = set_dates()
         RightsGranted.objects.create(
             basis=random.choice(RightsShell.objects.all()),
```

Comment:
Do you want to make sure that *all* the RightsShells have rights granted associated with them or is it useful for that to be randomized? Which option reflects the anticipated state of data in the system?

Reply:
If that's what you want to achieve, this test will achieve that. However, because you're selecting the RightShell you're attaching the RightsGranted to randomly, it's possible that you will end up less than 5 Rights Shells which have a RightsBasis, because you might grab the same one twice. Again, if that helps to mimic the shape of data you expect, that's fine.